src/core/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,152 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { getRandomValue, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/carousel.css';
+import { CarouselContext } from './CarouselContext';
+import { CarouselSlider } from './CarouselSlider';
+import { CarouselSlide } from './CarouselSlide';
+import { CarouselDotsList } from './CarouselDotsList';
+import { CarouselDot } from './CarouselDot';
+import { CarouselNavigation } from './CarouselNavigation';
+
+export type CarouselProps = {
+  /**
+   * Index of the currently shown slide.
+   * Can be used to set the default index or control the active slide programmatically.
+   * @default 0
+   */
+  activeSlideIndex?: number;
+  /**
+   * Callback fired when the current slide changes.
+   */
+  onSlideChange?: (index: number) => void;
+} & React.ComponentPropsWithoutRef<'div'>;
+
+/**
+ * The Carousel component consists of a set of slides, normally displayed one at a time. A navigation section is
+ * located below the slides, consisting of "dots" and "previous"/"next" buttons, used for changing slides.
+ *
+ * The currently shown slide can also be changed using the left/right arrow keys or by dragging on a touch device.
+ *
+ * This component uses a composition approach so it should be used with the provided subcomponents.
+ *
+ * @example
+ * <Carousel>
+ *   <Carousel.Slider>
+ *     <Carousel.Slide>...</Carousel.Slide>
+ *     <Carousel.Slide>...</Carousel.Slide>
+ *     <Carousel.Slide>...</Carousel.Slide>
+ *   </Carousel.Slider>
+ *   <Carousel.Navigation />
+ * </Carousel>
+ */
+export const Carousel = Object.assign(
+  React.forwardRef<HTMLElement, CarouselProps>((props, ref) => {
+    const {
+      activeSlideIndex: userActiveIndex = 0,
+      onSlideChange,
+      children,
+      ...rest
+    } = props;
+
+    // Generate a stateful random id if not specified
+    const [id] = React.useState(
```

Comment:
I need the value to stay the same between renders.

It's the same implementation as ComboBox. But I think it will not be stable between SSR and hydration, so we should come up with a better solution.

Reply:
Hopefully, we will be able to get rid of it
https://reactjs.org/docs/hooks-reference.html#useid